pyteal/ast/subroutine.py

```diff
@@ -374,15 +425,138 @@ def name(self) -> str:
         return self.subroutine.name()
 
     def type_of(self):
-        return self.subroutine.getDeclaration().type_of()
+        return self.subroutine.get_declaration().type_of()
 
     def has_return(self):
-        return self.subroutine.getDeclaration().has_return()
+        return self.subroutine.get_declaration().has_return()
 
 
 SubroutineFnWrapper.__module__ = "pyteal"
 
 
+class ABIReturnSubroutine:
+    """Used to create a PyTeal Subroutine (returning an ABI value) from a python function.
+
+    This class is meant to be used as a function decorator. For example:
+
+        .. code-block:: python
+
+            @ABIReturnSubroutine
```

Comment:
"Visual reminder" (as @michaeldiamant would say) to add this to our `user_guide_test()` (once the API is finalized)